### ISSUE TITLE
Notify users when new minor versions of clarinet are available

### ISSUE THREAD

**Issue by lgalabru** [stx-labs/clarinet#221]

**Comment by obycode** [comment]
Good idea. The GitHub API could be useful for implementing that - https://docs.github.com/en/rest/reference/releases#get-the-latest-release

**Comment by csgui** [comment]
Hello @lgalabru and @obycode! 

It seems that GitHub API needs a token to call the `latest release` endpoint. How that would be handled by clarinet? I mean, an authentication token will be need and the token information will be publicly available.

Probably there is something more secure to base this feature on (call to a private service, call to package managers, etc). Any idea on that? What you think?

Cheers!

**Comment by lgalabru** [comment]
hey @csgui! handling an API token in Clarinet could be a bit tedious.
How about relying on Brew's API?
https://formulae.brew.sh/api/formula/clarinet.json should do the work?

**Comment by csgui** [comment]
Yep @lgalabru ! Homebrew seems a reliable source of data for the need here. 

We can extract the latest released version from the map below:

```
"versions": {
    "stable": "1.3.0",
    "head": "HEAD",
    "bottle": true
}
```

If this task is aligned with clarinet priorities, I can be responsible to implement that. :-)

**Comment by lgalabru** [comment]
nice, that'd be helpful if we want to avoid version fragmentation + helping developers running the latest version.
There could be a relationship / dependency with this feature https://github.com/hirosystems/clarinet/issues/439, where we want to write some settings on disk (ideally respecting platform conventions so `.settings/clarinet.json` for linux, `Library/Application Data/clarinet` for macos, not sure for windows). 
We could start with a schema as simple as:
```json
{
   "update_check_enabled": true,
   "last_update_check": 1672070945
}
```
Hitting the brew API for every single command would be pretty annoying for developers and brew, and this preference file would be very useful for future features.

**Comment by csgui** [comment]
Probably also related with #205 where a TOML config file was discussed. What you think?

**Comment by lgalabru** [comment]
yes we could definitely go with a toml, no strong opinion on my end!

**Comment by obycode** [comment]
I noticed today that `gh` had a nice friendly way of doing this. I ran a command and after the output, it said:
```
A new release of gh is available: 2.29.0 → 2.30.0
To upgrade, run: brew upgrade gh
https://github.com/cli/cli/releases/tag/v2.30.0
```

[Their implementation](https://github.com/cli/cli/blob/e4e7bf5c8c47454f276e75773ef820c060cc3b5f/cmd/gh/main.go#L162) is in Go, but the process they used could be informative.

**Comment by hugo-stacks** [comment]
Replaced by #2276
